src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java

```diff
@@ -37,8 +37,12 @@
     extends AbstractViolationReporter
     implements FileSetCheck {
 
-    /** Collects the error messages. */
-    private final SortedSet<LocalizedMessage> messageCollector = new TreeSet<>();
+    /**
+     * Collects the error messages.
+     * @noinspection ThreadLocalNotStaticFinal
+     */
+    private final ThreadLocal<SortedSet<LocalizedMessage>> messageCollector =
```

Comment:
I'm not sure if I saw this inspection in other PR, please explain why this can't be static?

From IntelliJ:
> A non-static non-final
<b>java.lang.ThreadLocal</b> field associates state with
an instance-thread combination. This is seldom necessary and often a bug
which can cause memory leaks and possibly incorrect behavior.